src/components/FilterBar.ts

```diff
@@ -34,4 +45,35 @@ export default class FilterBar extends BaseComponent {
   private generateSortingOptions() {
     return JSON.stringify(KOREAN_SORT_FILTER);
   }
+
+  protected setEvent(): void {
+    document.addEventListener(
+      FILTER_EVENT.showFilter,
+      this.handleShowFilterBind
+    );
+  }
+
+  private handleShowFilter(e: Event) {
+    if (e instanceof CustomEvent) {
+      const { action } = e.detail;
+
+      if (action === ACTION_TYPES.open && !this.isDisplay) {
+        this.isDisplay = !this.isDisplay;
+        this.render();
+        return;
+      }
+
+      if (action === ACTION_TYPES.close || this.isDisplay) {
+        this.isDisplay = !this.isDisplay;
+        this.render();
+      }
+    }
+  }
```

Comment:
단순히 early return을 위해서라면 조건문을 반대로 주고 상위에서 return을 해주면 될거 같은데요 다르가 질문에서 의도하신 게 이게 맞는지 모르겠네요 😅

```suggestion
  private handleShowFilter(e: Event) {
    if (!(e instanceof CustomEvent)) return;

    const { action } = e.detail;

    if (action === ACTION_TYPES.open && !this.isDisplay) {
      this.isDisplay = !this.isDisplay;
      this.render();
      return;
    }

    if (action === ACTION_TYPES.close || this.isDisplay) {
      this.isDisplay = !this.isDisplay;
      this.render();
    }
  }
```